units/patch_intent.go

```diff
@@ -244,3 +252,144 @@ func (j *patchIntentProcessor) buildCliPatchDoc(patchDoc *patch.Patch, githubOau
 
 	return nil
 }
+
+func (j *patchIntentProcessor) buildGithubPatchDoc(patchDoc *patch.Patch, githubOauthToken string) (err error) {
+	adminSettings, err := admin.GetSettings()
+	if err != nil {
+		return errors.Wrap(err, "github pr testing is disabled, error retrieving admin settings")
+	}
+	if adminSettings.ServiceFlags.GithubPRTestingDisabled {
+		grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent), message.Fields{
+			"job":     patchIntentJobName,
+			"message": "github pr testing is disabled, not processing pull request",
+		})
+		return errors.New("github pr testing is disabled, not processing pull request")
+	}
+	defer j.Intent.SetProcessed()
+
+	mustBeMemberOfOrg := j.env.Settings().GithubPRCreatorOrg
+	if mustBeMemberOfOrg == "" {
+		return errors.New("Github PR testing not configured correctly; requires a Github org to authenticate against")
+	}
+
+	projectRef, err := model.FindOneProjectRefByRepo(patchDoc.GithubPatchData.BaseOwner,
+		patchDoc.GithubPatchData.BaseRepo)
+	if err != nil {
+		return errors.Wrapf(err, "Could not fetch project ref for repo '%s/%s'",
+			patchDoc.GithubPatchData.BaseOwner, patchDoc.GithubPatchData.BaseRepo)
+	}
+	if projectRef == nil {
+		return errors.Errorf("Could not find project ref for repo '%s/%s'",
+			patchDoc.GithubPatchData.BaseOwner, patchDoc.GithubPatchData.BaseRepo)
+	}
+
+	projectVars, err := model.FindOneProjectVars(projectRef.Identifier)
+	if err != nil {
+		return errors.Wrapf(err, "Could not find project vars for project '%s'", projectRef.Identifier)
+	}
+	if projectVars == nil {
+		return errors.Errorf("Could not find project vars for project '%s'", projectRef.Identifier)
+	}
+
+	isMember, err := authAndFetchPRMergeBase(patchDoc, mustBeMemberOfOrg,
+		patchDoc.GithubPatchData.Author, githubOauthToken)
+	if err != nil {
+		grip.Alert(message.Fields{
+			"message":  "github API failure: patch intents",
+			"job":      j.ID(),
+			"patch_id": j.PatchID,
+			"error":    err.Error(),
+		})
+		return err
+	}
+	if !isMember {
+		return errors.Errorf("user is not a member of %s", mustBeMemberOfOrg)
+	}
+
+	patchContent, err := fetchDiffByURL(patchDoc.GithubPatchData.DiffURL)
+	if err != nil {
+		return err
+	}
+
+	summaries, err := thirdparty.GetPatchSummaries(patchContent)
```

Comment:
I'm super dubious about any use of the thirdparty package

Reply:
GetPatchSummaries is how we run numstat against a patch. It's the same way the cli patch does it.

Reply:
it looks like this is only touching git commands, which I think is ok.